### DISCUSSION TITLE
Add Firefox Search Engine Doesn't work with searx/searx:latest Docker

### DISCUSSION THREAD

**Discussion by devn3t** [searx/searx#2192] [Installation issue: Docker]
I can't add my docker installation of Searx which is using http install (private) to my list of search engines in Firefox 80.

I've looked at https://github.com/searx/searx/issues/532 but it doesn't solve the issue because the default docker doesn't run apache.

I've added the base_url to my environment with http instead of https (http://192.168.1.145:8080) with no joy.  I've installed curl on the docker and was able to fetch the url with no issues.

Any idea why I'm still getting:

Firefox could not download the search plugin from:
http://192.168.1.145:8080/opensearch.xml

here is the output of the curl:

```
/usr/local/searx # curl -v http://192.168.1.145:8080/opensearch.xml
*   Trying 192.168.1.145:8080...
* Connected to 192.168.1.145 (192.168.1.145) port 8080 (#0)
> GET /opensearch.xml HTTP/1.1
> Host: 192.168.1.145:8080
> User-Agent: curl/7.69.1
> Accept: */*
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 200 OK
< Content-Type: application/opensearchdescription+xml; charset=utf-8
< Content-Length: 515
< Server-Timing: total;dur=8.095
< Connection: close
< 
<?xml version="1.0" encoding="utf-8"?>
<OpenSearchDescription xmlns="http://a9.com/-/spec/opensearch/1.1/">
  <ShortName>searx</ShortName>
  <Description>a privacy-respecting, hackable metasearch engine</Description>
  <InputEncoding>UTF-8</InputEncoding>
  <Image>/static/themes/oscar/img/favicon.png</Image>
  <LongName>searx metasearch</LongName>
    <Url rel="results" type="text/html" method="post" template="192.168.1.145">
        <Param name="q" value="{searchTerms}" />
    </Url>
* Closing connection 0
</OpenSearchDescription>
```

**Comment by dalf** [comment]
I'm sorry for the delay.

With:
```
make docker
docker run --rm -d -p 8888:8080 -e BASE_URL=$URL searx/searx
```

$URL:
* https://domain
* http://domain
* http://127.0.0.1:8888

Whatever $URL, I wasn't able to reproduce the bug you are describing.

Does it work with another browser Chrom* browser ?

(I'm not sure this docker issue)

**Comment by devn3t** [comment]
> 
> 
> I'm sorry for the delay.
> 
> With:
> 
> ```
> make docker
> docker run --rm -d -p 8888:8080 -e BASE_URL=$URL searx/searx
> ```
> 
> $URL:
> 
>     * https://domain
> 
>     * http://domain
> 
>     * http://127.0.0.1:8888
> 
> 
> Whatever $URL, I wasn't able to reproduce the bug you are describing.
> 
> Does it work with another browser Chrom* browser ?
> 
> (I'm not sure this docker issue)



Replaced with $URL and still no fix.  I can send you private details to access this if I port forward for a short time.

I don't have chrome installed to test but for search engine adding there, you can bypass and manually add things.

**Comment by dalf** [comment]
@devn3t you can send me an email (see on my profil).

**Comment by devn3t** [comment]
@dalf  Sent.

**Comment by dalf** [comment]
In settings.yml, for the [base_url](https://github.com/searx/searx/blob/21dbc7e852a5c48097c0067e78f0600972e54823/searx/settings.yml#L16) value

you have:
```yaml
  base_url: 420.420.420.420
```

add ```http://``` to have
```yaml
  base_url: http://420.420.420.420
```

so in opensearch.xml, instead of
```xml
<Url rel="results" type="text/html" method="post" template="420.420.420.420">
```
it will be:
```xml
<Url rel="results" type="text/html" method="post" template="http://420.420.420.420">
```

Related to https://github.com/searx/searx/issues/2048

**Comment by devn3t** [comment]
Pulled fresh image, created new contianer.  Changed all files as recommended in settings.yml and made sure it changed in both locations.

Restarted container.  Still no joy.

**Comment by devn3t** [comment]
So I've been able to add as a search engine now.  The issue was that the settings I was changing were not being set inside of settings.yml...they were instead writing to a file called settings.yml.new inside of /etc/searx/searx.

Once I placed my LOCAL_IP in the form of http://192.168.X.X inside of the settings.yml file and restarted, it wrote that settings.yml.new file inside of /etc/searx/searx.

So, I had to mv settings.yml.new settings.yml and restart the container before everything was read correctly.

**HOWEVER**

Search doesn't work on the newly added search engine.  I get 'unable to connect' and it looks like it's trying to reach port 80 and not port 8080...I'm thinking it doesn't like port 8080.  I'll do some more testing and report back here.

**Comment by devn3t** [comment]
Port changes aren't being honored...so I can't switch this to port 80.  Not sure if it's one of the underlying peices that are limiting this but if I change the port inside the settings.yml, it's not honoring this change.

Going to try to redeploy from scratch and use the URL + Port for the URL environmental variable.

**Comment by devn3t** [comment]
Ok, we're getting somewhere...I was able to recreate this from scratch with a new pull.  I added my IP URL to the URL env setting...I added my PORT on the end:  (http://192.168.X.X:8080) and I was able to add in the search engine.  If I don't do this, I'm unable to add it as a search engine.

Search still doesn't work using the newly added engine...when I search, it tries to connect to http**s**://192.168.X.X:8080 instead of what it's set at (http).

**Comment by FrostKnight** [comment]
Here is an odd thing, it does work in basilisk browser, but not palemoon at this moment. I discovered that recently, but forgot to submit this...

Weird it is... I am very much confused why the same web browser developers haven't fixed this bug yet... hmm, I wonder why that is the case...

**Comment by devn3t** [comment]
Checked in today and I can add this in to firefox now.  I'm running current as of the date this was posted.

Search still doesn't work...but I'll see what I can do to fix it.
